hil-test/tests/get_time.rs

```diff
@@ -20,10 +24,22 @@ impl Context {
 
         let delay = Delay::new(&clocks);
 
-        Context { delay }
+        Context {
+            delay,
+            #[cfg(esp32)]
+            clocks,
+        }
     }
 }
 
+fn time_moves_forward_during<F: FnOnce(Context)>(ctx: Context, f: F) {
+    let t1 = esp_hal::time::current_time();
+    f(ctx);
+    let t2 = esp_hal::time::current_time();
+
+    assert!(t2 > t1);
```

Comment:
Can the CPUs be so fast we end up with t2 == t1?

Reply:
I don't think so, no